fda-model/ead/simulation.cs

```diff
@@ -226,33 +233,105 @@ public void ComputeConditionalNonExceedanceProbability(paireddata.IPairedData fr
-            return stageDamage.compose(frequencyStage);
+            MeanRandomProvider meanRandomProvider = new MeanRandomProvider();
+            IPairedData frequencyStage = new PairedData(null,null);
+            IPairedData frequencyDamage = new PairedData(null, null, "Total");
+            IPairedData totalStageDamage = ComputeTotalStageDamage(_damage_category_stage_damage);
+            if (_levee_curve.IsNull)
+            {
+
+                if (_frequency_stage.IsNull)
+                {
+                    IPairedData frequencyFlow = BootstrapToPairedData(meanRandomProvider, _frequency_flow, 1000);
+                    if (_inflow_outflow.IsNull)
+                    {
+                        if (_flow_stage.IsNull)
+                        {
+                            throw new Exception("A rating curve must accompany a flow-frequency function");
+                        }
+                        else
+                        {
+                            IPairedData flowStageSample = _flow_stage.SamplePairedData(meanRandomProvider.NextRandom());
+                            frequencyStage = flowStageSample.compose(frequencyFlow);
+                        }
+                    }
+                    else
+                    {
+                        IPairedData inflowOutflowSample = _inflow_outflow.SamplePairedData(meanRandomProvider.NextRandom());
+                        IPairedData transformFlowFrequency = inflowOutflowSample.compose(frequencyFlow);
+                        if (_flow_stage.IsNull)
+                        {
+                            throw new Exception("A rating curve must accompany a flow-frequency function");
+                        }
+                        else
+                        {
+                            IPairedData flowStageSample = _flow_stage.SamplePairedData(meanRandomProvider.NextRandom());
+                            frequencyStage = flowStageSample.compose(transformFlowFrequency);
+                        }
+                    }
+
+                }
+                else
+                {
+                    frequencyStage = _frequency_stage.SamplePairedData(meanRandomProvider.NextRandom());
+                }
+
+                frequencyDamage = totalStageDamage.compose(frequencyStage);
+                double thresholdDamage = THRESHOLD_DAMAGE_PERCENT * frequencyDamage.f(THRESHOLD_DAMAGE_RECURRENCE_INTERVAL);
+                double thresholdStage = totalStageDamage.f_inverse(thresholdDamage);
+                return new Threshold(DEFAULT_THRESHOLD_ID, ThresholdEnum.InteriorStage, thresholdStage);
+            }
+            else
+            {
+                double topOfLevee = FindTopOfLevee(_levee_curve);
+                return new Threshold(DEFAULT_THRESHOLD_ID, ThresholdEnum.ExteriorStage, topOfLevee);
+            }
         }
 
-        private void ComputeDefaultThreshold()
+        internal double FindTopOfLevee(UncertainPairedData uncertainPairedData)
```

Comment:
what if there is no 1?

Reply:
There must be a coordinate with probability of failure = 1. This is the design in FDA 1 that we're carrying through. An exception is thrown in the case that a coordinate does not exist with prob of failure = 1.

Reply:
it is a percent chance of failure. there may be no 100% chance.  Or we let it go above the crest. i think using the percentage as a proxy for levee height is probably a mistake here. we might need to rethink that unfortunately. basically a levee object with an additional attribute of top of levee?